make/common/JdkNativeCompilation.gmk

```diff
@@ -76,18 +76,158 @@ endif
+      ifeq ($$(call isTargetOs, windows), true)
+        $1_$2_LIBPATH := $$(SUPPORT_OUTPUTDIR)/native/$$($1_$2_MODULE)/lib$$($1_$2_NAME)
+      else
+        ifeq ($(STATIC_LIBS), true)
+          $1_$2_LIBPATH := $$(SUPPORT_OUTPUTDIR)/native/$$($1_$2_MODULE)/lib$$($1_$2_NAME)/static
+        else
+          $1_$2_LIBPATH := $$(SUPPORT_OUTPUTDIR)/modules_libs/$$($1_$2_MODULE)
+        endif
+      endif
+    else
+      # Special treatment for hotspot
+      ifeq ($$(call isTargetOs, windows), true)
+        ifeq ($(STATIC_LIBS), true)
+          $1_$2_LIBPATH := $$(HOTSPOT_OUTPUTDIR)/variant-$$($1_$2_JVM_VARIANT_PATH)/libjvm/objs/static
+        else
+          $1_$2_LIBPATH := $$(SUPPORT_OUTPUTDIR)/modules_libs/$$($1_$2_MODULE)
+        endif
+      else
+        ifeq ($(CREATING_BUILDJDK), true)
+          # When building a buildjdk, it's always only the server variant
+          $1_$2_JVM_VARIANT_PATH := server
+        else
+          $1_$2_JVM_VARIANT_PATH := $(JVM_VARIANT_MAIN)
+        endif
+        ifeq ($(STATIC_LIBS), true)
+          $1_$2_LIBPATH := $$(HOTSPOT_OUTPUTDIR)/variant-$$($1_$2_JVM_VARIANT_PATH)/libjvm/objs/static
+        else
+          $1_$2_LIBPATH := $$(SUPPORT_OUTPUTDIR)/modules_libs/$$($1_$2_MODULE)/$$($1_$2_JVM_VARIANT_PATH)
+        endif
+      endif
+    endif
+  else
+    # Special treatment for virtual module "gtest"
+    ifeq ($$($1_$2_NAME), jvm)
+      ifeq ($$(call isTargetOs, windows), true)
+        ifeq ($(STATIC_LIBS), true)
+          $1_$2_LIBPATH := $$(JVM_OUTPUTDIR)/gtest/objs/static
+        else
+          $1_$2_LIBPATH := $$(JVM_OUTPUTDIR)/gtest/objs
+        endif
+      else
+        ifeq ($(STATIC_LIBS), true)
+          $1_$2_LIBPATH := $$(JVM_OUTPUTDIR)/gtest/objs/static
+        else
+          $1_$2_LIBPATH := $$(JVM_OUTPUTDIR)/gtest
+        endif
+      endif
+    else ifeq ($$($1_$2_NAME), gtest)
+      $1_$2_LIBPATH := $$(JVM_OUTPUTDIR)/libgtest
+      ifeq ($(STATIC_LIBS), true)
+        $1_$2_LIBPATH := $$(JVM_OUTPUTDIR)/libgtest/objs/static
+      else
+        $1_$2_LIBPATH := $$(JVM_OUTPUTDIR)/libgtest
+      endif
+    endif
+  endif
+endef
+
+# Process a JDK library designation such as "java.desktop:jawt" into the needed
+# additions to EXTRA_LINK_DEPS, LDFLAGS, LIBS, and EXTRA_HEADER_DIRS.
+# $1: The name of the rule (namespace)
+# $2: The safe namespace of the library
+# $3: The actual string given to represent the library
+define AddJdkLibrary
+  $1_$2_NAME := $$(strip $$(lastword $$(subst :, , $3)))
+  $1_$2_MODULE := $$(strip $$(patsubst %$$($1_$2_NAME), %, $3))
+
+  ifeq ($$($1_$2_MODULE), @:)
+    $1_$2_MODULE := $$(MODULE)
+  else ifeq ($$($1_$2_MODULE),)
+    $1_$2_MODULE := java.base
+  else
+    $1_$2_MODULE := $$(strip $$(patsubst %:, %, $$($1_$2_MODULE)))
+  endif
+
+  # Determine if the library in question is static.
+  ifeq ($(STATIC_BUILD), true)
+     $1_$2_STATIC_LIBRARY := true
+  endif
+  # Ideally, we should not hardcode these
```

Comment:
This is also something I plan to address in an upcoming PR. (The poor handling of static libraries is basically what has driven this current round of refactorings.)